test/unit_test/components/movement_sensor_test.dart

```diff
@@ -0,0 +1,285 @@
+      final result = await movementSensor.linearAcceleration();
+      expect(result.x, 0);
+    });
+
+    test('linearVelocity', () async {
+      final result = await movementSensor.linearVelocity();
+      expect(result.x, 0);
+    });
+
+    test('orientation', () async {
+      final result = await movementSensor.orientation();
+      expect(result.oX, 0);
+    });
+
+    test('position', () async {
+      final result = await movementSensor.position();
+      expect(result.altitude, 0);
+    });
+
+    test('properties', () async {
+      final result = await movementSensor.properties();
+      expect(result.angularVelocitySupported, false);
+    });
+
+    test('doCommand', () async {
+      final cmd = {'foo': 'bar'};
+      final resp = await movementSensor.doCommand(cmd);
+      expect(resp['command'], cmd);
+    });
+
+    test('extra', () async {
+      expect(movementSensor.extra, null);
+      await movementSensor.readings(extra: {'foo': 'bar'});
+      expect(movementSensor.extra, {'foo': 'bar'});
+    });
+  });
+
+  group('MovementSensor RPC Tests', () {
+    const String name = 'movementSensor';
+    late FakeMovementSensor movementSensor;
+    late ClientChannel channel;
+    late MovementSensorService service;
+    late Server server;
+
+    setUp(() async {
+      final port = 50000 + (name.hashCode % 10000);
+      movementSensor = FakeMovementSensor(name);
+      final ResourceManager manager = ResourceManager();
+      manager.register(MovementSensor.getResourceName(name), movementSensor);
+      service = MovementSensorService(manager);
+      channel = ClientChannel('localhost', port: port, options: const ChannelOptions(credentials: ChannelCredentials.insecure()));
+      server = Server([service]);
+      await server.serve(port: port);
+    });
+
+    tearDown(() async {
+      await channel.shutdown();
+      await server.shutdown();
+    });
+
+    group('MovementSensor Service Tests', () {
+      test('accuracy', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getAccuracy(GetAccuracyRequest(name: name));
+        expect(result.accuracy, {'accuracy': 0});
+      });
+
+      test('angularVelocity', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getAngularVelocity(GetAngularVelocityRequest(name: name));
+        expect(result.angularVelocity.x, 0);
+      });
+
+      test('compassHeading', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getCompassHeading(GetCompassHeadingRequest(name: name));
+        expect(result.value, 0);
+      });
+
+      test('linearAcceleration', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getLinearAcceleration(GetLinearAccelerationRequest(name: name));
+        expect(result.linearAcceleration.x, 0);
```

Comment:
again with this notation (result.linearAcceleration.x, 0), is this expecting result to call linearAcceleration passing 0 as parameter? what does the .x stand for?

Reply:
accessing value at result.linearAcceletation.x and comparing that value to 0

Reply:
yeah x is not a special keywords or anything, it is just the name of a member that we are accessing. and expecting its value to be `0`.